source/common/upstream/cluster_manager_impl.cc

```diff
@@ -211,7 +211,7 @@ ClusterManagerImpl::ClusterManagerImpl(const envoy::config::bootstrap::v2::Boots
   // Now setup ADS if needed, this might rely on a primary cluster.
   if (bootstrap.dynamic_resources().has_ads_config()) {
     ads_mux_.reset(new Config::GrpcMuxImpl(
-        bootstrap.node(),
+        local_info.node(),
```

Comment:
Can we add method similar to Config::Utility::checkLocalInfo("ads", local_info);  at https://github.com/envoyproxy/envoy/blob/master/source/common/config/grpc_mux_impl.cc#L18 and validate that node has right info?

Reply:
Yes, we can. However currently, if we failed to supply the info, it first throws an error for "cds", which I think serves the same purpose.

```
[2018-08-13 17:32:35.821][2350767][critical][main] source/server/server.cc:78] error initializing configuration '/envoy/config-local.yaml': cds: setting --service-cluster and --service-node is required
```

Reply:
@ramaraochavali after an experiment, I think it is tempting to have `Config::Utility::checkLocalInfo` inside the `GrpcMuxImpl`'s constructor (and testing it is easier, yeay!). However, since we want to pass in the `local_info` here, the implication is quite large (but mostly "replacing" `node` with `local_info`). I have the changeset ready but want to gather opinions about it. 

cc. @htuch @mattklein123.